### ISSUE TITLE
minifier should not move `super()` call to a class without super

### ISSUE THREAD

**Issue by kdy1** [swc-project/swc#5582]
### Describe the bug

Found by https://github.com/swc-project/swc/pull/5578

Minifier inlines `super()` calls and it breaks the code

### Input code

```typescript
"use strict";
class Hello {
  constructor() {
    return {
      toString() {
        return 'hello';
      },
    };
  }
}

class Outer extends Hello {
  constructor() {
    class Inner {
      [super()] = "hello";
    }

    return new Inner();
  }
}

expect(new Outer().hello).toBe('hello');
```

### Config

```json
{
  "jsc": {
    "parser": {
      "syntax": "ecmascript"
    },
    "target": "es2021",
    "minify": {
      "compress": {
        "arguments": false,
        "arrows": true,
        "booleans": true,
        "booleans_as_integers": false,
        "collapse_vars": true,
        "comparisons": true,
        "computed_props": true,
        "conditionals": true,
        "dead_code": true,
        "directives": true,
        "drop_console": false,
        "drop_debugger": true,
        "evaluate": true,
        "expression": false,
        "hoist_funs": false,
        "hoist_props": true,
        "hoist_vars": false,
        "if_return": true,
        "join_vars": true,
        "keep_classnames": false,
        "keep_fargs": true,
        "keep_fnames": false,
        "keep_infinity": false,
        "loops": true,
        "negate_iife": true,
        "properties": true,
        "reduce_funcs": false,
        "reduce_vars": false,
        "side_effects": true,
        "switches": true,
        "typeofs": true,
        "unsafe": false,
        "unsafe_arrows": false,
        "unsafe_comps": false,
        "unsafe_Function": false,
        "unsafe_math": false,
        "unsafe_symbols": false,
        "unsafe_methods": false,
        "unsafe_proto": false,
        "unsafe_regexp": false,
        "unsafe_undefined": false,
        "unused": true,
        "const_to_let": true,
        "pristine_globals": true
      },
      "mangle": false
    },
    "loose": false
  },
  "minify": false
}
```

### Playground link

https://play.swc.rs/?version=1.2.241&code=H4sIAAAAAAAAA4WOvQ7CMAyE9zyFlaWJhPoCFQsTTAyMiAEFC5Aqp0octRLKu5M%2Fqm5stu%2FO38ngETy7t2E5CDPevYcjjqOFjwAwlpIWDFundLkAOOTgqC0AbC8pTc9V33i6V%2F7UDe0ed2WIeY8iioY7B0YHuDDS4x%2B8Jk5EKfHDXX2YMFlusAdZiLISE2FThnCuQaXXArhMaFhlqbRQui8PdM%2F2gKr1T%2F4vZQPuiCcBAAA%3D&config=H4sIAAAAAAAAA22TS3LjMAxE76K1F5Msc4C5BgsmQRkZflQE6FiVyt0HtCxKsrNTPwEg2AC%2Fh0%2B2w8f3MEFhLO2L5yRwGz4GtBHYFppk%2BDkNAmVEaZjf%2F7y%2FDachUiI%2FtxSb41SQuX1rWI2YRIWHwHhSUvKXSilV1TnngJBetAE2lARHLFuqzSHAxGiuUHpGOw0KcU4HVAWdmUqedjQ5EspJi63MIThjs8MOqKAVuuIWojU0JLE21ju5Q4fnOo7NpiUSrxAqSK%2BFt7sNemLPu2RiMb4mfkKHThe0XPIRRd4UlFrSGvOZKR18%2BIeofQZgThBxy7xzr3M4Rvpfoih5HaLMnYa86yrhqJczRL7fsHWNRWhzq6CrFtsN7Vb8AQ8XYnJo0Ht1uyfzF4m9bMVknjD7LtU18NsMFmnWfTrStgIv8K92JftxPHgEuTwznuM5h5cSEeWS3QtWIyQ%2Fw6Lbe5ueaU0O1WZ0ux%2BVm%2BxrqsOXbEJ7X6vPuhCaY8aQz3199RlGSGNfy5%2F7wHgn1ze56P%2F0pMps3gMAAA%3D%3D

### Expected behavior

`super()` should not throw

### Version

1.2.241

**Comment by iheyunfei** [comment]
It seems fixed in [1.3.24](https://play.swc.rs/?version=1.3.24&code=H4sIAAAAAAAAA4WOvQ7CMAyE9zyFlaWJhPoCFQsTTAyMiAEFC5Aqp0octRLKu5M%2Fqm5stu%2FO38ngETy7t2E5CDPevYcjjqOFjwAwlpIWDFundLkAOOTgqC0AbC8pTc9V33i6V%2F7UDe0ed2WIeY8iioY7B0YHuDDS4x%2B8Jk5EKfHDXX2YMFlusAdZiLISE2FThnCuQaXXArhMaFhlqbRQui8PdM%2F2gKr1T%2F4vZQPuiCcBAAA%3D&config=H4sIAAAAAAAAA22TS3LjMAxE76K1F5Msc4C5BgsmQRkZflQE6FiVyt0HtCxKsrNTPwEg2AC%2Fh0%2B2w8f3MEFhLO2L5yRwGz4GtBHYFppk%2BDkNAmVEaZjf%2F7y%2FDachUiI%2FtxSb41SQuX1rWI2YRIWHwHhSUvKXSilV1TnngJBetAE2lARHLFuqzSHAxGiuUHpGOw0KcU4HVAWdmUqedjQ5EspJi63MIThjs8MOqKAVuuIWojU0JLE21ju5Q4fnOo7NpiUSrxAqSK%2BFt7sNemLPu2RiMb4mfkKHThe0XPIRRd4UlFrSGvOZKR18%2BIeofQZgThBxy7xzr3M4Rvpfoih5HaLMnYa86yrhqJczRL7fsHWNRWhzq6CrFtsN7Vb8AQ8XYnJo0Ht1uyfzF4m9bMVknjD7LtU18NsMFmnWfTrStgIv8K92JftxPHgEuTwznuM5h5cSEeWS3QtWIyQ%2Fw6Lbe5ueaU0O1WZ0ux%2BVm%2BxrqsOXbEJ7X6vPuhCaY8aQz3199RlGSGNfy5%2F7wHgn1ze56P%2F0pMps3gMAAA%3D%3D).

**Comment by kdy1** [comment]
Thank you!
